nasl/nasl_packet_forgery.c

```diff
@@ -1437,7 +1438,16 @@ nasl_send_packet (lex_ctxt *lexic)
       bzero (&sockaddr, sizeof (struct sockaddr_in));
       sockaddr.sin_family = AF_INET;
       sockaddr.sin_addr = sip->ip_dst;
-      if (sockaddr.sin_addr.s_addr != inaddr.s_addr)
+
+      if (allow_broadcast)
+        {
+          if (setsockopt (soc, SOL_SOCKET, SO_BROADCAST, &i, sizeof (i)) < 0)
```

Comment:
I think it would be better to use a variable in this block instead of i, which was defined way up above and is not very descriptive.